### PR TITLE
Refactored the code in release-notes page slugs

### LINKED ISSUE #6964: Refactor the code in Release notes page slugs
## Description
 $subject.
Refactor the code by adding existing reusable components.
Eg:- https://github.com/ballerina-platform/ballerina-dev-website/blob/master/pages/learn/integration/%5B...slug%5D.js

### PR DESCRIPTION
**Pull request by JayPokale**
## Purpose
Refactor the code
> Fixes #6964

## Checklist
- [x] **Page restrcuture**
- components/common/main-content/MainContent.js
- pages/downloads/swan-lake-release-notes/[...slug].js
- pages/downloads/0.9.x-release-notes/[...slug].js
- pages/downloads/1.0.x-release-notes/[...slug].js
- pages/downloads/1.1.x-release-notes/[...slug].js
- pages/downloads/1.2.x-release-notes/[...slug].js

### CODE REVIEW

File: components/common/main-content/MainContent.js [hunk 1]
```diff
@@ -267,3 +347,6 @@ export default function MainContent(props) {
     </ReactMarkdown>
   );
 }
+
+// content -> data
+// codes -> codeSnippets
```
Comment (sm1990) [inline components/common/main-content/MainContent.js:352]: Remove this line

File: components/common/main-content/MainContent.js [hunk 2]
```diff
@@ -16,29 +16,62 @@
...
-}
+};
 
 export default function MainContent(props) {
-
   const content = props.content;
   const codes = props.codes ? new Map(JSON.parse(props.codes)) : new Map();
 
+  // Synatax highlighting
+  const HighlightSyntax = (code, language) => {
```
Comment (sm1990) [inline components/common/main-content/MainContent.js:44]: We do not need to do any changes in this file. Also the changes you have done creates rendering issues in some pages which uses this component.

Can you revert the changes of this file?

Comment (JayPokale) [reply components/common/main-content/MainContent.js:44]: By removing this, syntax highlighting will be disabled in some pages. By using conditions, it can be fixed.
Can you please try updated code in new PR.

File: pages/downloads/0.9.x-release-notes/[...slug].js [hunk 3]
```diff
@@ -19,14 +19,10 @@
...
 import React from "react";
-import ReactMarkdown from "react-markdown";
 import { Container, Col, Button, Offcanvas } from "react-bootstrap";
-import remarkGfm from "remark-gfm";
-import rehypeRaw from "rehype-raw";
 import Image from "next-image-export-optimizer";
 import Head from "next/head";
-import { getHighlighter, setCDN } from "shiki";
-
+import { setCDN } from "shiki";
```
Comment (sm1990) [inline pages/downloads/0.9.x-release-notes/[...slug].js:25]: Remove `setCDN`.  and `setCDN("https://unpkg.com/shiki/");` in line 31. There are not used anymore. Check other slugs too.

File: components/common/main-content/MainContent.js (hunk 2 above)
Comment (sm1990) [reply components/common/main-content/MainContent.js]: Looks like you were right. We need that function to add syntax highlighting in pages where the `html`s are used as below. I am sorry I missed that scenario.
image

But your previous fix (with changes to mainContent.js) gave an error in this page https://ballerina.io/learn/generate-code-documentation/
There is a nested code block which is what causing the issue I think.
Can you check that?

Comment (JayPokale) [reply components/common/main-content/MainContent.js:44]: I observed that only `ballerina` language is used in this function. All other language code comes from `props.codes`.
I guess I can remove all other languages from this function.
If I keep other languages too then it is giving some error in other file.
HighlightSyntax function

Please visit the new PR on this issue.

Comment (sm1990) [reply components/common/main-content/MainContent.js:44]: Getting an error. 
image

Comment (JayPokale) [reply components/common/main-content/MainContent.js]: It was working fine in my device. I added try cache block for any possible error due to this function.
Please try new PR once.

Comment (sm1990) [reply components/common/main-content/MainContent.js]: @JayPokale I think the main issue is we have some old release notes in `html` format and I don't think we can re-use the `mainComponent.js` as expected. I think we have to go with your original fix (https://github.com/ballerina-platform/ballerina-dev-website/pull/8067/files). I tested that again. But noticed that the headers & the toc were missing in that fix.
image

**Comment by JayPokale** [comment]
https://github.com/ballerina-platform/ballerina-dev-website/pull/8071/commits/a0915cab22d47882cfdb1b36e0536481173e078e This code working fine after use of try cache block. Please try this once.
I am also trying another method for the same issue.

File: components/common/main-content/MainContent.js (hunk 2 above)
Comment (JayPokale) [reply components/common/main-content/MainContent.js]: New code works fine due to try cache block.
I am also trying another method for the same issue.
I also observed one issue. If you are offline then HighlightSyntax giving fetching some error. I fixed this too in new PR.
Please check my new PR.

Comment (sm1990) [reply components/common/main-content/MainContent.js]: Getting this error now. 
image

Did you check your old fix (https://github.com/ballerina-platform/ballerina-dev-website/pull/8067/files) as well?

That nested code block related issue will be tracked via https://github.com/ballerina-platform/ballerina-dev-website/issues/6656

Comment (JayPokale) [reply components/common/main-content/MainContent.js]: Now I fixed it by old method

Comment (JayPokale) [reply components/common/main-content/MainContent.js]: Please test my new PR. I used old fix in this.

**Comment by sm1990** [comment]
Sorry about the delay. I tested your latest fix. It still gives an error.
image
Shall we revert the changes in `mainContent`?

**Comment by JayPokale** [comment]
> Sorry about the delay. I tested your latest fix. It still gives an error. image Shall we revert the changes in `mainContent`?

Yeah, I will revert changes in `mainContent`.
Can you please send the steps to recreate this error as I am trying to recreate same but not getting.
It looks like this project uses some .wasm file which is not loading properly.

File: components/common/heading/RenderHeading [hunk 4]
```diff
@@ -0,0 +1,79 @@
+import React from "react";
```
Comment (sm1990) [inline components/common/heading/RenderHeading:1]: Add the file extension `.js`.

Add the license statement at the top. You can get it from [here](https://github.com/ballerina-platform/ballerina-dev-website/blob/ace4e0e7efcad1503bd5ad0462e1b0998cd8b34d/components/common/footer/Footer.js#L1). change the year to 2023.

Comment (JayPokale) [reply components/common/heading/RenderHeading:19]: License Statement added in `RenderHeading` component in new PR.

Comment (JayPokale) [reply components/common/heading/RenderHeading:19]: License Statement added in `RenderHeading` component in new PR.

Comment (sm1990) [reply components/common/heading/RenderHeading:19]: @JayPokale Add the file extension `.js` to the filename

Comment (JayPokale) [reply components/common/heading/RenderHeading]: Sorry for this mistake. It's now corrected.

**Review by sm1990 — APPROVED** [review]

**Comment by sm1990** [comment]
@JayPokale, Thank you for your contribution. We are currently hosting a `Ballerina Hacktoberfest`. Since this issue is eligible for our event, please visit https://ballerina.io/hacktoberfest/#rules & submit the form. 😄
